### DISCUSSION TITLE
Fan curve lower RPM than 1800

### DISCUSSION THREAD

**Discussion by ivca33** [BartoszCichecki/LenovoLegionToolkit#1013] [Q&A] [answered]
(I'm first time using LegionToolkit, I have bought new laptop after 12 years, also I'm first time on github, so please sorry if I'm not posting this like I should)

Can someone tell me please if there is a way to set RPM on fan curve below 1800, but not to turn them off. 
For example, to set 1600rpm for first blue dot/circle on first slide column.
I have seen that this is possible on LegionFanControl,  so maybe it is also here, but I don't know how.

The thing is that I don't want to turn fans completely off, but then again above 1600rpm there is some resonance fan whistling sound.
(there is a lot of mention of these phenomena on Reddit, it is not coil whine, because when fans are off there is zero noise. It is a well recorded design flaw. Fans spinning near steel cover mesh and produce a whistling sound. If I put two 1" wide 2" long tapes, on outer cover, across fans, in specific position, so actually close something like 10% air intake, sound disappear. People actually do this. I Also saw that Lenovo on newer models have attached some flat plastic parts, exactly in this spot on this mesh, inside of laptop, probably  to close some of the intake holes because of this. I can't see any other reason). 

Because I mostly use the laptop for simple tasks, I rarely need cooling above 1600rpm, so this would be very convenient.

Anyway, it is not something extra important, but why not ask, right? Thank you very much

**Comment by CoMb0BrEaKeR** [comment]
From what I can tell it's not possible, those settings are behind the API provided by Lenovo so the LegionToolkit cannot make changes.  Maybe someone else can chime in?  I also would like more granular fan control and have the fans turn on at lower temps and at lower speed.  These computers are beasts on AC power but sometimes you actually need to use it on battery and they can run HOT even in lowest power modes.

**Comment by ivca33** [comment]
@CoMb0BrEaKeR, I was thinking the same, but I did try "LegionFanControl", and there it is possible to change it to all RPM speeds. I have tried 1600 and 1300, and it worked, "LegionToolkit" have confirmed speeds. But I don't like LegionFanControl and in the end I have uninstall it. Didn't respond every time like it should. Maybe it was my fault, but it doesn't matter. LegionToolkit is much nicer.

Also, I did mange to solve my fan whistle problem manually, so now I don't need any more optional fan curve anymore.
I have put some filter pads in specific position outside on the back cover, betwean grils and there is no whistle noise anymore. I don't even know that fans when they start are working haha, I'm very happy costumer now.

Here is picture, maybe helps someone:
![mojsmall](https://github.com/BartoszCichecki/LenovoLegionToolkit/assets/149418634/37f9bc56-d596-47a3-b2ba-28847ff6763d)

Also, here is a original thread on Reddit where people talk about the whistle problem, and fixes, also pictures how Lenovo did it, if someone is more interested.
https://www.reddit.com/r/GamingLaptops/comments/nvs5xv/lenovo_5_2021_mass_problem_high_frequency_noise/

Anyway, back to topic. If there is possibility for more precise fan control it would be nice, but also like someone have mentioned for  "LegionFanControl" (don't know where I have read it) maybe is not safe to do it like they do, so maybe it is better to leave this in LenovoLegionToolkit like it is.

**Reply by BartoszCichecki** [reply, marked as answer]
There is no possibility for more fan control using Lenovo methods that LLT uses. More fine grained control is only possible using "the LFC method" which is directly modifying EC registers, which is neither safe nor reliable.
